File: account/src/main/java/com/example/account/exception/AccountException.java
package com.example.account.exception;

public class AccountException extends Exception {
    private final ExceptionCode code;

    public AccountException(ExceptionCode code) {
        super(code.getMessage());
        this.code = code;
    }

    public AccountException(ExceptionCode code, String message) {
        super(message);
        this.code = code;
    }

    public ExceptionCode getCode() {
        return code;
    }
}

File: services/account/account-api/src/main/kotlin/com/example/account/api/rest/exception/AccountException.kt
package com.example.account.api.rest.exception

class AccountException(var code: ExceptionCode): Exception(code.message)

enum class ExceptionCode(val message: String) {
    UNKNOWN("Unknown"),
    BAD_REQUEST("Bad request"),
    ACCOUNT_NOT_FOUND("Account not found"),
    ACCOUNT_EXISTS("AccountModel already exists"),
    ACCOUNT_NAME_EXISTS("AccountModel name already exists"),
    ROLE_NOT_FOUND("RoleModel not found");
}